llvm/lib/CodeGen/TargetLoweringObjectFileImpl.cpp

```diff
@@ -2645,10 +2653,14 @@ MCSection *TargetLoweringObjectFileXCOFF::getSectionForTOCEntry(
     const MCSymbol *Sym, const TargetMachine &TM) const {
   // Use TE storage-mapping class when large code model is enabled so that
   // the chance of needing -bbigtoc is decreased.
+  // The "_$TLSML" symbol for TLS local-dynamic mode requires XMC_TC.
```

Comment:
You can use addis/load to load _$TLSML.  In this case, you can use XMC_TE.

Reply:
When I change from XMC_TC to XMC_TE in the large code model like following,
```
L..C0:  
        .tc _Renamed..5f24__TLSML[TE],_Renamed..5f24__TLSML[TE]@ml
        .rename _Renamed..5f24__TLSML[TE],"_$TLSML"
```
The assembler complains like following:
```
Assembler:
tls_ld_te.s: line 62: 1252-215 The ML relocation type is only valid for a reference
        to a TOC symbol from the symbol itself.
```
Change back to "TC" did not see the message.

Reply:
Looks like I have to keep the XMC_TC setting to make the assembler happy?

Reply:
You're right.  I didn't know the assembler would complain.